File: tests/fixtures/typescript/src/service.ts
/** Calls, references, annotations and a boundary. */

import { Base, Engine, type Identifier } from "@/store";

export const DEFAULT_REGION: string = "eu";

export function build(region: string): Engine {
  const engine = new Engine(region);
  engine.ping();
  return engine;
}

export function inspect(item: Base): string {
  return item.describe();
}

export async function fetchOne(ident: Identifier): Promise<string> {
  const response = await fetch(`/engines/${ident}`);
  return response.text();
}

File: tests/fixtures/python/pkg/service.py
"""Calls, references, annotations and a boundary."""

import httpx

from pkg.models import Base, Engine, Identifier


def build(region: str) -> Engine:
    engine = Engine(region)
    engine.ping()
    return engine


def inspect(item: Base) -> str:
    return item.describe()


def fetch_one(ident: Identifier) -> str:
    response = httpx.get(f"/engines/{ident}")
    return response.text
